src/software/simulated_tests/simulated_test_fixture.py

```diff
@@ -247,6 +267,8 @@ def run_test(
         :param tick_duration_s: length of a tick
         :param index: index of the current test. default is 0 (invariant test)
                       values can be passed in during aggregate testing for different timeout durations
+        :param run_till_end: If true, test runs till the end even if eventually validation passes
```

Comment:
what do you think about this being a 5 second buffer after all eventually validation passes. so it's the minimum of timeout and 5 seconds after all the eventually validation passes?

Reply:
hmm, cuz from talking to Nima, he mentioned how some tests, we just want to specify a time for it to last and see what happens in the test. For those tests, we would have empty eventually validation which will "pass" vacuously instantly. So the runner will choose between the given timeout and 0 + 5 seconds, so we won't be able to run these tests for more than 5 seconds